libuavcan_drivers/stm32/driver/src/uc_stm32_can.cpp

```diff
@@ -1091,6 +1097,38 @@ int CanDriver::init(const uavcan::uint32_t bitrate, const CanIface::OperatingMod
     return res;
 }
 
+int CanDriver::configureIface(const uavcan::uint32_t bitrate, const CanIface::OperatingMode mode, uavcan::uint8_t can_number)
+{
+    int res = 0;
+
+    UAVCAN_STM32_LOG("Initing iface %i...", can_number);
+
+    if (can_number == 0) {
+        /*
+        * CAN1
+        */
+        ifaces[0] = &if0_;                          // This link must be initialized first,
+        res = if0_.init(bitrate, mode);             // otherwise an IRQ may fire while the interface is not linked yet;
+    }
+#if UAVCAN_STM32_NUM_IFACES > 1
+    else if (can_number == 1) {
+        /*
+        * CAN2
+        */
+        ifaces[1] = &if1_;                          // Same thing here.
+        res = if1_.init(bitrate, mode);
+    }
+#endif
```

Comment:
Per MISRA, every condition that contains an else-if should have a plain else clause at the end. In this case, it must handle the case when `can_number` is neither 0 nor 1. There are several other occurrences where this should be updated.

Reply:
how about a switch statement with an empty default?

Reply:
If you find that reasonable, sure (in my opinion `if` works well here)

Reply:
nevermind, it gets messy